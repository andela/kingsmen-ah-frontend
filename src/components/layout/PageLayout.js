import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Header from '@components/commons/Header';


function PageLayout(props) {
  const { children } = props;
  return (
    <Fragment>
      <Header />

      {children}
    </Fragment>
  );
}

PageLayout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.array
  ]).isRequired
}

export default PageLayout;