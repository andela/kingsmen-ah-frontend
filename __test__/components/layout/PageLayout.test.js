import React from 'react';
import PageLayout from '@components/layout/PageLayout';


describe('<PageLayout />', () => {
  it('should display the page layout correctly', () => {
    const wrapper = shallow(
      <PageLayout>
        <div>Hello World</div>
      </PageLayout>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('PageLayout')).toBeTruthy();
  });

  it('should display the children props properly', () => {
    const wrapper = shallow(
      <PageLayout>
        <div>Hello</div>
      </PageLayout>
    );
    
    expect(wrapper.props().children).toBeTruthy()
  })
});
