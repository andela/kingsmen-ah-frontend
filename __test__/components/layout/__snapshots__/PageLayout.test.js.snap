// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLayout /> should display the page layout correctly 1`] = `
<Fragment>
  <Connect(withRouter(Header)) />
  <div>
    Hello World
  </div>
</Fragment>
`;
